website/server/libs/spells.js

```diff
@@ -163,14 +163,32 @@ async function castSpell (req, res, { isV3 = false }) {
       task: results[1],
     });
   } else if (targetType === 'self') {
-    await castSelfSpell(req, user, spell, quantity);
-
-    let userToJson = user;
-    if (isV3) userToJson = await userToJson.toJSONWithInbox();
-
-    res.respond(200, {
-      user: userToJson,
-    });
+    // Check if chilling frost or stealh skill has been previously casted or not.
+    // See #12361 for more details.
+    const spellName = spell.key;
+    const incompleteDailiesDue = await Tasks.Task.find({
+      userId: user._id,
+      type: 'daily',
+      $and: [
+        {type: 'daily', completed: false, isDue: true },
+      ],
+    }).exec();
+    if (spellName === 'frost' && user.stats.buffs.streaks) {
+      throw new BadRequest(res.t('spellWizardFrostAlreadyCast'));
+    }
+    else if (spellName === 'stealth' && user.stats.buffs.stealth >= incompleteDailiesDue.length) {
```

Comment:
can we run the query to calculate the number of incomplete dailies only when stealth is being used?

so that it's not run un-necessarily when using other spells

Reply:
Yes, this would be better. The query will now run only when `Stealth` is cast.